### ISSUE TITLE
Using an XML schema to convert a document to JSON fails to always use a list in case of nested sequences

### ISSUE THREAD

**Issue by ztane** [sissaschool/xmlschema#162]
I had a given XML schema that had a nested sequence; essentially it has an equivalent of:

```
      <xs:sequence>
        <xs:sequence maxOccurs="unbounded">
          <xs:element name="object" type="objType"/>
        </xs:sequence>
      </xs:sequence>
```

when using 

```
import xmlschema
from pprint import pprint

xs = xmlschema.XMLSchema('collection.xsd')
pprint(xs.to_dict('collection.xml'))
```

`to_dict` does not understand to use a *list* for a single `<object>`; the code produces

```
{'@xmlns:col': 'http://example.com/ns/collection',
 '@xmlns:xsi': 'http://www.w3.org/2001/XMLSchema-instance',
 '@xsi:schemaLocation': 'http://example.com/ns/collection collection.xsd',
 'object': {'@available': True,
            '@id': 'b0836217462',
            'author': {'@id': 'PAR',
                       'born': '1841-02-25',
                       'dead': '1919-12-03',
                       'name': 'Pierre-Auguste Renoir',
                       'qualification': 'painter'},
            'estimation': Decimal('10000.00'),
            'position': 1,
            'title': 'The Umbrellas',
            'year': '1886'}}
```

It works as expected if the `maxOccurs="unbounded"` is in the outer `<xs:sequence>` though. Now this is a bit unfortunate since I need to rewrite the XML schema file by flattening the nested sequences before using it for JSON serialization.

**Comment by ztane** [comment]
The used files (slightly modified collection.xsd + collection.xml from the test suite)

[collection.xml](https://github.com/sissaschool/xmlschema/files/4063958/collection.xml.txt)
[collection.xsd](https://github.com/sissaschool/xmlschema/files/4063959/collection.xsd.txt)

**Comment by brunato** [comment]
Hi,
it's the internal method *ParticleMixin.is_single()* that determines the choice to put also single items in a list:

https://github.com/sissaschool/xmlschema/blob/7d1ff4f6172715b24f92ce1da4fa3baa8d97e1e1/xmlschema/validators/xsdbase.py#L924-L925

Substituting this method for model groups (*ModelGroup* class in *models.*py) with:

```python
    def is_single(self):
        if self.max_occurs != 1 or not self:
            return False
        elif len(self) > 1:
            return True
        else:
            return self[0].is_single()
```
the condition should be more reliable.

Thanks

**Comment by brunato** [comment]
Available in v1.1.0. Please close this issue if it's solved.
Bye
